sqlitecluster/SQLiteNode.h

```diff
@@ -205,10 +213,32 @@ class SQLiteNode : public STCPNode {
     chrono::steady_clock::time_point _lastNetStatTime;
 
     // Handler for transaction messages.
-    void handleBeginTransaction(Peer* peer, const SData& message);
-    void handleCommitTransaction(Peer* peer, const SData& message);
-    void handleRollbackTransaction(Peer* peer, const SData& message);
+    void handleBeginTransaction(SQLite& db, Peer* peer, const SData& message);
+    void handleCommitTransaction(SQLite& db, Peer* peer, const uint64_t commandCommitCount, const string& commandCommitHash);
+    void handleRollbackTransaction(SQLite& db, Peer* peer, const SData& message);
 
     WallClockTimer _syncTimer;
-    uint64_t _handledCommitCount;
+    atomic<uint64_t> _handledCommitCount;
+
+    // List of threads that are handling replication. These are transient and only exist while FOLLOWING.
+    list<thread> _replicationThreads;
+
+    // State variable that indicates when the above threads should quit.
+    atomic<bool> _replicationThreadsShouldExit;
+
+    // Mutex that guards the map of hashes for transactions currently being replicated.
+    mutex _replicationHashMutex;
+    map<uint64_t, string> _replicationHashes;
+
+    // Mutex, condition variable, and queue object for command 
+    mutex _replicationCommandMutex;
+    condition_variable _replicationCV;
+    SSynchronizedQueue<pair<Peer*, SData>> _replicationCommands;
+
+    // Set of hashes for commands that need to be rolled back. This re-uses the _replicationHashMutex as a guard for
+    // changes. This is seldom used.
+    set<string> _rollbackHashes;
```

Comment:
I suggest renaming this to `_replicationHashesToRollback`

Reply:
I'd also suggest moving this definition up to be right next to `_replicationHashesToCommit`